tests/ClientTest.php

```diff
@@ -345,11 +346,12 @@ public function testAddConnectionWithSslTransportAndForceIpv4ThrowsException()
 
         try {
             $this->client->addConnection($connection);
-            $this->fail('Expected exception was not thrown');
-        } catch (Exception $e) {
             $this->assertSame(Exception::ERR_CONNECTION_STATE_UNSUPPORTED, $e->getCode());
+        } catch (Exception $e) {
+            $this->fail('Unexpected exception was thrown');
         }
     }
+*/
```

Comment:
I am uncertain how to test this and that is why I commented it out.
The actual socket for Secure Connections is created in TcpConnecter (react/socket-client) and as far as I can see cannot be requested to check the context options.

I did find I could get the context options using:
stream_context_get_options($connection->getData('stream'))

However getData is not setup in MockConnection.

Could you help me on this one?